packages/blade/src/components/Amount/Amount.tsx

```diff
@@ -0,0 +1,270 @@
+   */
+  intent?: Exclude<Feedback, 'neutral'>;
+  /**
+   * Sets the size of the amount.
+   *
+   * @default 'body-medium'
+   */
+  size?:
+    | `title-${NonNullable<Exclude<TitleProps['size'], 'large'>>}`
+    | `heading-${NonNullable<Exclude<HeadingProps<{ variant: 'regular' }>['size'], 'medium'>>}`
+    | `heading-${NonNullable<
+        Exclude<HeadingProps<{ variant: 'regular' }>['size'], 'medium'>
+      >}-${NonNullable<Exclude<TextProps<{ variant: 'body' }>['weight'], 'regular'>>}`
+    | `body-${NonNullable<Exclude<TextProps<{ variant: 'body' }>['size'], 'xsmall'>>}`
+    | `body-${NonNullable<Exclude<TextProps<{ variant: 'body' }>['size'], 'xsmall'>>}-${NonNullable<
+        Exclude<TextProps<{ variant: 'body' }>['weight'], 'regular'>
+      >}`;
+  /**
+   * Indicates what the suffix of amount should be
+   *
+   * @default 'decimals'
+   */
+  suffix?: 'decimals' | 'none' | 'humanize';
+  /**
+   * Makes the prefix symbol and decimal digits small and faded
+   *
+   * @default true
+   */
+  isAffixSubtle?: true | false;
+  /**
+   * Prefix to be shown before the amount value. The prefix can be either a currency symbol or a currency code.
+   *
+   * @default 'currency-symbol'
+   */
+  prefix?: 'currency-symbol' | 'currency-code';
+  /**
+   * The currency of the amount.
+   *
+   * @default 'INR'
+   * */
+  currency?: Currency;
+} & TestID &
+  StyledPropsBlade;
+
+type ColorProps = {
+  amountValueColor: BaseTextProps['color'];
+  affixColor: BaseTextProps['color'];
+};
+
+const getTextColorProps = ({ intent }: { intent: AmountProps['intent'] }): ColorProps => {
+  const props: ColorProps = {
+    amountValueColor: 'surface.text.normal.lowContrast',
+    affixColor: 'surface.text.muted.lowContrast',
+  };
+  if (!intent && intent !== 'neutral') return props;
+  props.amountValueColor = `feedback.text.${intent}.lowContrast`;
+  props.affixColor = `feedback.text.${intent}.lowContrast`;
+  return props;
+};
+
+interface AmountValue extends Omit<AmountProps, 'value'> {
+  affixColor: BaseTextProps['color'];
+  amountValueColor: BaseTextProps['color'];
+  value: string;
+  size: Exclude<AmountProps['size'], undefined>;
+}
+
+const useIsNative = (): boolean => {
+  return getPlatformType() === 'react-native';
+};
+
+const AmountValue = ({
+  value,
+  size,
+  amountValueColor,
+  isAffixSubtle,
+  suffix,
+  affixColor,
+}: AmountValue): ReactElement => {
+  const affixFontWeight = isAffixSubtle ? 'regular' : 'bold';
+  const affixFontSize = isAffixSubtle ? affixFontSizes[size] : amountFontSizes[size];
+  const valueForWeight = size.includes('bold') || size.startsWith('title') ? 'bold' : 'regular';
+  const isNative = useIsNative();
+  if (suffix === 'decimals' && isAffixSubtle) {
+    const integer = value.split('.')[0];
+    const decimal = value.split('.')[1];
+
+    // Becasue Basetext doesn't support display flex we need to wrap it in a fragment for web
+    const AmountWrapper = isNative ? BaseText : React.Fragment;
```

Comment:
Why do we need a variable that calls a function that calls another function to do just one check? Use it directly?
```suggestion
    const AmountWrapper = getPlatformType()==='react-native' ? BaseText : React.Fragment;
```